usi/src/main/java/daq/usi/BaseSwitchController.java

```diff
@@ -0,0 +1,187 @@
+package daq.usi;
+
+/*
+ * Licensed to Google under one or more contributor license agreements.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public abstract class BaseSwitchController implements Runnable, SwitchController {
+  /**
+   * Terminal Prompt ends with '#' when enabled, '>' when not enabled.
+   */
+  public static final String CONSOLE_PROMPT_ENDING_ENABLED = "#";
+  public static final String CONSOLE_PROMPT_ENDING_LOGIN = ">";
+
+  // Define Common Variables Required for All Switch Interrogators
+  protected SwitchTelnetClientSocket telnetClientSocket;
+  protected Thread telnetClientSocketThread;
+  protected String remoteIpAddress;
+  protected int telnetPort;
+  protected boolean debug;
+  protected String username;
+  protected String password;
+  protected boolean userAuthorised = false;
+  protected boolean userEnabled = false;
+  protected String hostname = null;
+  protected boolean commandPending = false;
+
+  public BaseSwitchController(String remoteIpAddress, int telnetPort, String username,
+                              String password) {
+    this(remoteIpAddress, telnetPort, username, password, false);
+  }
+
+  /**
+   * Abstract Switch controller. Override this class for switch specific implementation
+   *
+   * @param remoteIpAddress switch ip address
+   * @param telnetPort      switch telnet port
+   * @param username        switch username
+   * @param password        switch password
+   * @param debug           for verbose logging
+   */
+  public BaseSwitchController(
+      String remoteIpAddress, int telnetPort, String username, String password, boolean debug) {
+    this.remoteIpAddress = remoteIpAddress;
+    this.telnetPort = telnetPort;
+    this.username = username;
+    this.password = password;
+    this.debug = debug;
+    telnetClientSocket =
+        new SwitchTelnetClientSocket(remoteIpAddress, telnetPort, this, debug);
+  }
+
+  protected boolean containsPrompt(String consoleData) {
```

Comment:
just checking -- so this code (contains prompt) is standard for both AT and Cisco?

Reply:
Yes. It’s also the same on my home switch actually